vspace/src/main/java/edu/asu/diging/vspace/web/general/search/PublicSearchController.java

```diff
@@ -0,0 +1,195 @@
+     * This also filters modules which are linked to the spaces.
+     * 
+     * @param modulePagenum current page number sent as request parameter in the URL.
+     * @param model         This the object of Model attribute in spring MVC.
+     * @param searchTerm    This is the search string which is being searched.
+     */
+    private void paginationForModule(String modulePagenum, Model model, String searchTerm) {
+        Page<IModule> modulePage = staffSearchManager.searchInModules(searchTerm, Integer.parseInt(modulePagenum));
+        model.addAttribute("moduleCurrentPageNumber", Integer.parseInt(modulePagenum));
+        model.addAttribute("moduleTotalPages", modulePage.getTotalPages());
+        List<ModuleWithSpace> moduleList = new ArrayList<>();
+        
+        //Adding space info for each module
+        for(IModule module : modulePage.getContent()) {
+            ModuleLink moduleLink = moduleLinkManager.findFirstByModule(module);
+            if(moduleLink!=null) {
+                ModuleWithSpace modWithSpace = new ModuleWithSpace();
+                try {
+                    BeanUtils.copyProperties(modWithSpace, module);
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    logger.error("Could not create moduleWithSpace.", e);
+                }
+                modWithSpace.setSpaceId(moduleLink.getSpace().getId());
+                moduleList.add(modWithSpace);
+            }
+        }
+        model.addAttribute("moduleSearchResults", moduleList);
+        model.addAttribute("moduleCount", modulePage.getTotalElements());
+    }
+
+    /**
+     * This method is used to search the search string specified in the input
+     * parameter(searchTerm) and return the slides corresponding to the page number
+     * specified in the input parameter(spacePagenum) whose name or description
+     * contains the search string. The slidePagenum is starting from 1. 
+     * This also filters Slides from modules which are linked to the spaces.
+     * 
+     * @param slidePagenum current page number sent as request parameter in the URL.
+     * @param model        This the object of Model attribute in spring MVC.
+     * @param searchTerm   This is the search string which is being searched.
+     */
+    private void paginationForSlide(String slidePagenum, Model model, String searchTerm) {
+        Page<ISlide> slidePage = staffSearchManager.searchInSlides(searchTerm, Integer.parseInt(slidePagenum));
+        model.addAttribute("slideCurrentPageNumber", Integer.parseInt(slidePagenum));
+        model.addAttribute("slideTotalPages", slidePage.getTotalPages());
+        List<ISlide> slideList = new ArrayList<>();
+        Set<String> slideSet = slideManager.getAllSlidesFromStartSequences();
+        
+       //Adding space info for each slide
+        for(ISlide slide : slidePage.getContent()) {
+            if(slideSet.contains(slide.getId())) {
```

Comment:
what is this condition for?

Reply:
the search result for slides will list only those slides which are part of a start sequence. Is this not required?